sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java

```diff
@@ -375,9 +376,16 @@ private synchronized void loginFromUI() {
             mAuthenticationClient.acquireToken(mActivity, new String[]{mIdentityScope}, new AuthenticationCallback() {
 
                 @Override
-                public void onSuccess(IAuthenticationResult authenticationResult) {
+                public void onSuccess(final IAuthenticationResult authenticationResult) {
                     AppCenterLog.info(LOG_TAG, "User login succeeded. id=" + authenticationResult.getIdToken());
-                    // TODO send id token (not access token) to ingestion, if sdk still enabled.
+                    getInstance().post(new Runnable() {
+                        @Override
+                        public void run() {
+                            if (isEnabled().get()) {
```

Comment:
post ensures its enabled already, please remove that check
everything is executed in a sequence so state cannot change (its the background thread looper).